apps/bot/utils.ts

```diff
@@ -73,3 +83,119 @@ export const replyWithEmbedError = (
     ],
   })
 }
+
+export const LockPostWithReason = async (
+  interaction: ChatInputCommandInteraction,
+  reason: string,
+) => {
+  if (!interaction.channel?.isThread()) {
+    await replyWithEmbedError(interaction, {
+      description: 'This command can only be used in a thread/forum post',
+    })
+    return
+  }
+
+  const mainChannel = interaction.channel.parent
+  if (mainChannel && mainChannel.type === ChannelType.GuildForum) {
+    const lockedTagId = mainChannel.availableTags.find((t) =>
+      t.name.includes('Locked'),
+    )?.id
+
+    if (lockedTagId) {
+      const newTags = Array.from(
+        new Set([...interaction.channel.appliedTags, lockedTagId]),
+      )
+      interaction.channel.setAppliedTags(newTags)
+    }
+  }
+
+  await interaction.reply({ content: 'Ok!', ephemeral: true })
+
+  await interaction.channel.setLocked(true)
+  await interaction.channel.send({
+    embeds: [
+      {
+        color: Colors.Blue,
+        title: '🔒 Post Locked',
+        description: reason,
+      },
+    ],
+  })
+  await unindexPost(interaction.channel)
+}
+
+export const modifyRegularMemberRoles = async (
+  interaction: ChatInputCommandInteraction,
+  shouldAddPoints: boolean,
+) => {
+  const user = interaction.options.getUser('user', true)
+
+  const guildMember = await interaction.guild?.members.fetch(user.id)
+
+  if (!guildMember) {
+    await interaction.reply({
+      content: "I couldn't find the guild member from this user",
+      ephemeral: true,
+    })
+    return
+  }
+
+  await interaction.deferReply({ ephemeral: true })
+
+  await syncUser(user, guildMember)
+  if (shouldAddPoints) {
+    await addFullPointsToUser(user.id)
+  } else {
+    await removeFullPointsFromUser(user.id)
+  }
+  await tryToSetRegularMemberRole(guildMember, true)
+
+  await interaction.editReply({ content: 'Done!' })
+}
+
+export const checkInvalidAnswer = async (
```

Comment:
This name seems to better describe what the function does 
```suggestion
export const validateForumChannel = async (
```

Reply:
Done 👍🏼